forge-gui/src/main/java/forge/player/PlayerControllerHuman.java

```diff
@@ -2849,10 +2850,24 @@ private void addCardToZone(ZoneType zone, final boolean repeatLast, final boolea
             getGame().getAction().invoke(() -> {
                 if (targetZone == ZoneType.Battlefield) {
                     if (!forgeCard.getName().equals(f.getName())) {
-                        forgeCard.changeToState(forgeCard.getRules().getSplitType().getChangedStateName());
-                        if (forgeCard.getCurrentStateName().equals(CardStateName.Transformed) ||
-                                forgeCard.getCurrentStateName().equals(CardStateName.Modal)) {
-                            forgeCard.setBackSide(true);
+                        if (forgeCard.getRules().getSplitType().equals(CardSplitType.Specialize)) {
+                            if (f.getName().equals(forgeCard.getState(CardStateName.SpecializeW).getName())) {
+                                forgeCard.changeToState(CardStateName.SpecializeW);
+                            } else if (f.getName().equals(forgeCard.getState(CardStateName.SpecializeU).getName())) {
+                                forgeCard.changeToState(CardStateName.SpecializeU);
+                            } else if (f.getName().equals(forgeCard.getState(CardStateName.SpecializeB).getName())) {
+                                forgeCard.changeToState(CardStateName.SpecializeB);
+                            } else if (f.getName().equals(forgeCard.getState(CardStateName.SpecializeR).getName())) {
+                                forgeCard.changeToState(CardStateName.SpecializeR);
+                            } else if (f.getName().equals(forgeCard.getState(CardStateName.SpecializeG).getName())) {
+                                forgeCard.changeToState(CardStateName.SpecializeG);
+                            }
```

Comment:
`getSpecializeParts()` would give you a better map to iterate over all the CardStates and CardFaces